test/circt-reduce/trivial.sh

```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+! firtool "$1" 2>&1 | grep "error: sink \"x1.x\" not fully initialized" >/dev/null
```

Comment:
Suggestion for future patches: I'd recommend making this a bit more realistic, it is sort of a "test-error.sh" script and you can make the error message come in from $2.  Alternatively you could have the `circt-reduce` have a few builtin modes:

1) -run-script foo.sh
2) -reduce-error "error: sink ..."
3) -reduce-crash

This will make writing tests for circt-reduce easier and will eliminate the need to write/find/copy the trivial scripts.

Reply:
I love this suggestion! I've tried to stick with what `mlir-reduce` does for no particular reason. But in practice I'm pretty sure you can avoid writing a custom script by just having an option that regex-matches the output. Also, it would probably make sense to have the tester/command be a positional argument that gobbles up everything after the input file, such that you can do `circt-reduce foo.mlir -- firtool --verilog --lower-to-hw --some-magic-flag`.

Reply:
Opened #1602 to track this.